packages/shared/src/telemetry/clientCache.ts

```diff
@@ -0,0 +1,97 @@
+import type { TelemetryEventRaw } from './types';
+
+type TtlInMilliseconds = number;
+
+const DEFAULT_CACHE_TTL_MS = 1000; // 24 hours
+
+/**
+ * Manages caching for telemetry events using the browser's localStorage to
+ * mitigate event flooding in frequently executed code paths.
+ */
+export class TelemetryClientCache {
+  #storageKey = 'clerk_telemetry';
+  #cacheTtl = DEFAULT_CACHE_TTL_MS;
+
+  cacheAndRetrieve(event: TelemetryEventRaw): boolean {
```

Comment:
if we change the naming this could be: `isEventThrottled`

Reply:
Yeap this name feels more appropriate, my understanding here is that if a key needs to be invalidated we just delete and expect the next attempt to cache it again